### DISCUSSION TITLE
[Feature Proposal] - Generate a GitGraph from a repository?

### DISCUSSION THREAD

**Discussion by sneakers-the-rat** [mermaid-js/mermaid#4884] [Ideas]
I have searched the issues and discussions for gitgraph, as well as searched the web more broadly for this and was sort of surprised not to find anything - would the maintainers be interested in a simple cli script for generating GitGraphs from a repository? Not sure if this goes here or in `mermaid-cli`, but i think that would be super useful and doesn't seem that challenging.

Something like 

```bash
mermaid-cli generate gitgraph
  --path              # Path to git repo (otherwise cwd)
  --commits           # Range of commits, if a single short hash, use that as the start and graph until present. If `hash..hash` use as a range
  --include-branches  # specific branches to include, excluding all others, otherwise include all.
  --exclude-branches  # specific branches to exclude, including all others, otherwise include all.
  --no-tags           # don't draw tags
  --config            # options to configure whole graph
  --commit-config     # options to configure display of a specific commit
  --output            # output file!
```

and then output the mermaid to stdout if no output is provided, or render if the filetype is .svg or .png

**Comment by wolfspyre** [comment]
THIS would be astoundingly cool!! 

especially love that  it includes an option to exclude a specific branch  
(like the branch it lives in)  
so as to be able to be generated  automagically via CI

**Reply by sneakers-the-rat** [reply]
alright honestly i hadn't even thought about using it in CI but that makes uh a lot of sense and so i'll write this tomorrow bc now i want it lol
